### ISSUE TITLE
Add subscription model functionality to licenses

### ISSUE THREAD

**Issue by imetandy** [snipe/snipe-it#5673]
**Server (please complete the following information):**
 - Snipe-IT Version - 4.4.1
 - OS: [e.g. Ubuntu, CentOS] - Ubuntu 16.04 Server
 - Web Server: [e.g. Apache, IIS] - Nginx
 - PHP Version - PHP7.0

**Is your feature request related to a problem? Please describe.**
I manage licenses that have different models for payments. Some are one off costs (plugins etc) and some are on a subscription basis, like the Adobe suite or Office 365. 
While we are able to track our spend, I would have to submit a new license every month to track the ongoing cost of retaining that license. 

**Describe the solution you'd like**
I would like to have a selector option for renewal time to put against each license. 
**e.g. list:** 
One off cost (non recurring)
Monthly
Yearly
Quarterly

This would then calculate the yearly spend for reports, as well as allowing us to track costs more efficiently. 

**E.g.** 
License A - £350 - One off cost - Purchased 1/1/2018
License B - £100 - Monthly - purchased 1/1/2018

**Yearly software spend:**
350 + (100*12) = £1,550

**Describe alternatives you've considered**
Previously I tracked this in a spreadsheet and had a similar setup. One cell would have the listed options above (monthly, yearly, quarterly) and this would feed into a formula based on the option multiplying the cost by how many times it would recur in a year. e.g. 12 times for monthly, 52 times for weekly, 4 times for quarterly. 

**Additional context**

**Comment by joepitt91** [comment]
+1

**Comment by pmzhou** [comment]
+1
Most licenses are now monthly or yearly, so this would be handy.

**Comment by sjlaird** [comment]
+1 Yes please.

**Comment by cwcage** [comment]
+1
I have just started trying to use Snipe-IT and I ran into this exact issue fairly quickly. Every few months I seem to have a request for another software or service that is a subscription of some sort. Usually it is a per-user subscription paid monthly or annually. 

I was really hoping I'd be able to track subscription costs and was already trying to figure out how to make it work when I found this issue report.

**Comment by tliderek** [comment]
+1

**Comment by imetandy** [comment]
Hey @snipe , can this be labelled as a feature request please? 

Thanks! 👍

**Comment by liceng** [comment]
+1

**Comment by mduesi** [comment]
+1

**Comment by sjlaird** [comment]
I still want this

**Comment by mduesi** [comment]
It is important. We are using many cloud subscriptions and we would like to manage all those in Snipe. Thanks for taking care of this request.

**Comment by tymik** [comment]
+1 for sure! more and more timed licenses on the market, less lifetime ones. definitely it's a must!

**Comment by keiranlovett** [comment]
+1

**Comment by jmlassen** [comment]
+1

**Comment by wRx7MSTi** [comment]
+1

**Comment by pbrink231** [comment]
Just ran into this issue. +1

Basically need to add
- checkbox for recurring
- monthly, yearly
- start date (exists already)
- recuring date (when normal billing starts to run)
- pre prorated
- post prorated

When a subscription starts it might not start recurring till the 1st so they pro rate the first month.  Also when ending is the same thing.

When you want to change the seats in my mind would end the previous license and make a new clone.  You either post prorate the previous license or pre prorate the new license for the seat changes.

That should cover all bases.  This should also work for assets where you rent assets and have a monthly with pre and post prorated terms.

**Comment by MartinGF** [comment]
> 
> 
> **Server (please complete the following information):**
> 
>     * Snipe-IT Version - 4.4.1
> 
>     * OS: [e.g. Ubuntu, CentOS] - Ubuntu 16.04 Server
> 
>     * Web Server: [e.g. Apache, IIS] - Nginx
> 
>     * PHP Version - PHP7.0
> 
> 
> **Is your feature request related to a problem? Please describe.**
> I manage licenses that have different models for payments. Some are one off costs (plugins etc) and some are on a subscription basis, like the Adobe suite or Office 365.
> While we are able to track our spend, I would have to submit a new license every month to track the ongoing cost of retaining that license.
> 
> **Describe the solution you'd like**
> I would like to have a selector option for renewal time to put against each license.
> **e.g. list:**
> One off cost (non recurring)
> Monthly
> Yearly
> Quarterly
> 
> This would then calculate the yearly spend for reports, as well as allowing us to track costs more efficiently.
> 
> **E.g.**
> License A - £350 - One off cost - Purchased 1/1/2018
> License B - £100 - Monthly - purchased 1/1/2018
> 
> **Yearly software spend:**
> 350 + (100*12) = £1,550
> 
> **Describe alternatives you've considered**
> Previously I tracked this in a spreadsheet and had a similar setup. One cell would have the listed options above (monthly, yearly, quarterly) and this would feed into a formula based on the option multiplying the cost by how many times it would recur in a year. e.g. 12 times for monthly, 52 times for weekly, 4 times for quarterly.
> 
> **Additional context**
> NA
> 
> Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/59359298-add-subscription-model-functionality-to-licenses?utm_campaign=plugin&utm_content=tracker%2F505106&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F505106&utm_medium=issues&utm_source=github).

This would be awesome

**Comment by pfeigl** [comment]
This is our most wanted feature with so much cloud subscriptions happening today.

**Comment by tylerwel** [comment]
+1

**Comment by AngelusDulcis** [comment]
+1

**Comment by snickbit** [comment]
+1

**Comment by dlwasylyk** [comment]
++1

**Comment by adicaprio** [comment]
I would love to have this feature... at this time you can only set a start and end date.  Having an option for autorenewal (daily/weekly/monthly/yearly) would be great!

**Comment by carloshmiranda** [comment]
+1

**Comment by b12e** [comment]
+1

**Comment by RichardBouwman** [comment]
After working with an CMDB Excel sheet we just started using Snipe-it.

Even though it's 100% better then the Excel sheet, we love to see the annual subscription option in the license tab! 

++1

**Comment by m4rphel** [comment]
+1

**Comment by JeroenV99** [comment]
+1

**Comment by StarWhiz** [comment]
+1

It would awesome to be able to keep track of who the admins of each SAAS applications are too!

It's not much but here's a bounty https://www.bountysource.com/issues/59359298-add-subscription-model-functionality-to-licenses

**Comment by dops84** [comment]
+1

**Comment by coryjt** [comment]
+1 for robust subscription options. Ideally with the capability of tracking if different licenses have varying renewal dates.

**Comment by orrywrightrfm** [comment]
+1 and for the ability to be able to be able to specify if subs are monthly or yearly.

**Comment by b12e** [comment]
Yes it is

**Comment by kerbje** [comment]
Wondering if this functionality/license model is going to make into v5?
Definitely a +1

**Comment by joviwap** [comment]
+1

**Comment by orrywrightrfm** [comment]
Still a +1 here.

**Comment by kylemorrow** [comment]
+1

**Comment by Shbleh** [comment]
+1

**Comment by arielromeo** [comment]
+1

**Comment by cronjevh** [comment]
+1

**Comment by sarang2502** [comment]
+1

**Comment by Relix360** [comment]
+1

**Comment by TalHawk** [comment]
+1 This is an essential feature for licensing

**Comment by ChristophAuch** [comment]
+1

**Comment by kerbje** [comment]
Can we label this issue as a feature request?

**Comment by snipe** [comment]
This has always been on the roadmap. Subscriptions weren't as common when Snipe-IT was first created, wasn't really set up that way, but we have obviously seen them get a lot more popular, so we know this is necessarily functionality. The way Licenses work in general all be changing after v5 is out, which is why we haven't tried to roll this into v5.

**Comment by ghost** [comment]
Heyo @snipe! Any update on when this might be rolled out?

**Comment by nianios** [comment]
Any news?

**Comment by mbnetops** [comment]
We could really use this!!!

**Comment by emumovies** [comment]
Just ran into this issue, kind of gutted it's missing +100 is this coming?

**Comment by bro1188** [comment]
+1

**Comment by shawn-wm** [comment]
+1

**Comment by robinostlund** [comment]
+1

**Comment by orrywrightrfm** [comment]
+1 because it would be nice but after 3 years of complete silence I highly doubt this is ever going to happen.

**Comment by Ng-marelec** [comment]
+1 This is a must in any license management tool!
Also a history of any sorts would be handy in view of cost management.
EG. MS365-Monthly: january might have 250 licenses, february 240, march 245 etc...

**Comment by pmzhou** [comment]
I have been waiting for this feature for years but there's no response. They could at least make it a premium feature. In the meantime I have been using Spiceworks' Track Your Vendor Contracts<https://apps.spiceworks.com/tools/contracts/> which does the job(well kind of).
Hope this helps someone.


________________________________

**Comment by sjlaird** [comment]
Sorry to say we had to abandon Snipe-IT in the end due to it being evident that this feature is not coming. We offered to pay for it to be developed but this was rejected. I like the product but a the end of the day it just did not fulfill our requirement for licensing. I'm unsubbing now but @snipe, I wish you and the team all the best for future development.

**Comment by SufiSR** [comment]
+1

**Comment by jburns-1315** [comment]
+1 My company has many licenses that are subscription, just like most other companies these days. This is a must for license tracking, any update from the devs would be appreciated, even if it is to tell us that this feature isn't coming.

**Comment by OwainD** [comment]
I'm not sure there is much software these days that isn't on a subscription model, I don't really understand how this functionality isn't in Snip It yet. I'll have to look at a different solution as this doesn't appear to be on their radar for some reason. Seems pretty fundamental these days.

**Comment by snipe** [comment]
Still very much on our radar.

**Comment by mparksGP** [comment]
+1

While we are extremely happy with Snipe for physical asset tracking, the software industry as a whole is moving away from fixed cost licenses, rendering the license tracking feature as it exists today pretty useless for anything other than operating systems and a small handful of other big name server software.

**Comment by Tyree** [comment]
I'm finding this being "on the radar" for over 4 years disconcerting. I'm just starting to use Snipe-IT and adding our licenses in. There definitely needs to be more options for the terms of the licenses. Folks have mentioned subscriptions and monthly, annually, etc. But we also have licenses that renew every 2 or 3 years. It'd be great to be able to define these terms so that we can see a true cost per time period.
I came searching on github to figure out how to best use the licenses feature. I added 13 seats at a cost of $250 as that's the cost of all 13 licenses, but then Snipe is multiplying the cost by the number of seats. Definitely need an option for license packs as dividing the cost by 13 manually comes up with 19.2307692... That's not gonna work. :-D
I'm not sure how best to implement a more robust license system, but there are some smart folks working on Snipe. I'm confident they can come up with something awesome. Thanks!

**Comment by nik-wolfe** [comment]
+1 for this. Almost all licensing is now procured on a recurring basis rather than a one off cost.

**Comment by a8bit** [comment]
Just started setting this up yesterday and immediately hit the Microsoft 365 problem. Is there any word when/if this is going to be implemented?

**Comment by viclou** [comment]
+1 33726
Possibly trigger alerts monthly for assets based on termination date if recurring box is selected

**Comment by yeco** [comment]
Any word on this feat?

**Comment by jbryant88** [comment]
This would be incredibly handy feature! Is there any word on if this will be coming out in an update? @snipe

**Comment by rdagger** [comment]
Just pitched Snipe-It to our management and the first question was does it support subscriptions.

**Comment by jbryant88** [comment]
> Just pitched Snipe-It to our management and the first question was does it support subscriptions.

We love Snipe! But the only fee reporting that is shows is a Yearly cost, and If you put the full cost that you pay, it shows up with that cost for each individual. Really hoping that Snipe response and modifies the function so you can put in that its a Monthly Subscription or A Yearly etc.

**Comment by SufiSR** [comment]
Dear snipe time, almost another year has gone by since your last response.
In 2020, it was said, you would want to tackle this when V5 is out. I understand, plans change and more important things may come up. However, there are no real workarounds available and in today's world, one-off licensing is rare - subscriptions are standard. Next up is usage-based.

Is this still on your radar for implementation?

**Comment by Tyree** [comment]
> Dear snipe time, almost another year has gone by since your last response. In 2020, it was said, you would want to tackle this when V5 is out. I understand, plans change and more important things may come up. However, there are no real workarounds available and in today's world, one-off licensing is rare - subscriptions are standard. Next up is usage-based.

Agreed. The lack of a subscription licensing feature makes the licenses functionality of Snipe-IT far less useful. The work-arounds are very confusing and really have no way of telling the real story of cost, expiration, etc. As much as I loathe subscriptions, they aren't going anywhere and have spread into even the smallest of applications. What used to be $40, one-time, is now $5/month forever. :-D

**Comment by snipe** [comment]
> Is this still on your radar for implementation?

It is. We have to rewrite the way licenses currently work before we can tackle that though.

**Comment by untold-jw** [comment]
+1

**Comment by delpechp** [comment]
Just started using Snipe-IT and a little frustrated it doesn't have subscription model for licensing is this coming soon??
@snipe

**Comment by benaiahhenry** [comment]
This is really needed.  I just implemented Snipe-IT a couple weeks ago, and was disappointed to see how limited the software tracking is.  We just started started using our first usage based product (AutoDesk Flex) and it would be great to be able to track that as well as our Office365, and VOIP subscriptions on Snipe...

**Comment by eddiekelley** [comment]
Being able to track subscriptions would be a killer feature!

**Comment by 2ndSysAdmin** [comment]
I've been working with Snipe-IT for a while and am a little frustrated that there is no subscription model (monthly cost) for licensing.
Can you estimate when it will be implemented?

**Comment by 2ndSysAdmin** [comment]
It wouldn't be bad if you could assign a license to another license. To cover purchase and maintenance. We have this case in many places and currently have no solution other than how to assign the license to each user separately, which is actually not acceptable.

**Comment by nbj-nine** [comment]
+1
We need this badly, and we also need a way to track Cloud Assets

**Comment by wiggywoosaact** [comment]
+1 subscriptions models are the norm

**Comment by ryrun** [comment]
+1 yes please. But also with price history, so we can see price increase.

**Comment by emumovies** [comment]
It’s been 6 years, we gave up on Snipe. I think the project has been dead since 2021.

**Comment by snipe** [comment]
@emumovies  Yes, the project has been dead, that's why we're still developing features every week, thousands of commits and dozens of releases since then. 🙄

If you gave up on Snipe-IT then unsubscribe from the thread and stay off the repo. There's enough negativity in the world without your 💩 comments.

**Comment by emumovies** [comment]
Apologies, I just checked the website on my phone.  That was the last feature update.  Wasn’t trying to be rude.  I stayed subscribed to see if this feature would ever be added.  Didn’t mean to trigger, will gracefully unsubscribe.

**Comment by snipe** [comment]
@emumovies It's still very much something we want to build, but we have to basically rewrite how licenses work on an architectural level, while also going back in history (via the action logs) to make sure history isn't mangled. The licenses architecture is something that's been on deck for a while, but because what we do could beak a lot of people's workflows, we have to be extremely *careful* about how we do it and make sure we don't accidentally erase history.

Cloud licensing/reworking licenses is the first priority (along with custom fields on other first class items) once v7 is out.

**Comment by pereira-ax** [comment]
HI @snipe 
Any updates on adding this feature in snipe-it??

**Comment by snipe** [comment]
Not yet, but soon. We're currently reworking licenses to account for cloud, recurring licenses, long-term licenses whose prices can change, etc. It's a big project, and we need to make sure we're backwards compatible, so it's kind of a lot. We're being slow and careful, but I promise, we haven't lost sight of this one.

**Comment by Tyree** [comment]
> Not yet, but soon. We're currently reworking licenses to account for cloud, recurring licenses, long-term licenses whose prices can change, etc. It's a big project, and we need to make sure we're backwards compatible, so it's kind of a lot. We're being slow and careful, but I promise, we haven't lost sight of this one.

Thanks for your hard work! It definitely does not sound like a trivial task!

**Comment by kenjones007** [comment]
> Not yet, but soon. We're currently reworking licenses to account for cloud, recurring licenses, long-term licenses whose prices can change, etc. It's a big project, and we need to make sure we're backwards compatible, so it's kind of a lot. We're being slow and careful, but I promise, we haven't lost sight of this one.

Thanks for the update! I appreciate that you're taking this to the next level and considering all these aspects carefully. I fully agree—better to get it later but comprehensive rather than a quick shot that might not cover everything properly. Looking forward to seeing the improvements when they’re ready!

**Comment by mugen6** [comment]
> Not yet, but soon. We're currently reworking licenses to account for cloud, recurring licenses, long-term licenses whose prices can change, etc. It's a big project, and we need to make sure we're backwards compatible, so it's kind of a lot. We're being slow and careful, but I promise, we haven't lost sight of this one.

Just my opinion, but instead of reworking the way you do licenses, you could leave the existing code alone and just add a new feature called "subscriptions". That way you wouldn't have to worry about backwards compatibility because you would have complete separation.

**Comment by snipe** [comment]
@mugen6 unfortunately, we do have to revamp licenses based on how people use them. It’s still very much a priority, I we just have to be super careful so we don’t mangle history.

**Comment by mugen6** [comment]
@snipe Thank you, but that is not relevant to my point. Adding a new feature under a new section called _subscriptions_ would not have to impact licenses or its history at all. It can be entirely separate and licenses as it is could remain.

**Comment by emumovies** [comment]
I am just amazed that this thread is 7 years old.  No movement, but still responding.

**Comment by snipe** [comment]
> Thank you, but that is not relevant to my point.

@mugen6 > if it effectively duplicate cloud/subscription behavior, then it is pretty relevant IMHO.

> I am just amazed that this thread is 7 years old. No movement, but still responding.

@emumovies - and this comment was constructive how? We work on 100 things at a time. This one just happens to be more difficult than other requests.

**Comment by snipe** [comment]
For anyone still interested in this topic, the epic has been open for some time. Make your cases there. https://github.com/grokability/snipe-it/issues/17546 - I've explained our planned approach there.
